dev-python/*: version bumps

arthurzam opened the pull request:
A Pull Request with various version bumps for python packages. This is continuation of mailing with 
@mgorny .

1. All were tested on `python2_7`, `python3_7`, `pypy`, `pypy3` and passed successfully.
2. All were bumped to `EAPI=7`
3. I have tried to correct `BDEPEND`. I know that `dev-python/setuptools` should be a `BDEPEND`, but not sure about others like `dev-python/pytest`, `dev-python/nose`. What should be done about test dependencies?
4. When `IUSE="test"` I added to new `ebuilds`: `RESTRICT="!test? ( test )"`.
5. One point of concern is `KEYWORDS=amd64-linux`. `dev-lang/python` doesn't support this keyword, but a lot of `dev-python/*` packages set it.

mgorny commented:
Without reviewing yet, few notes:
1. Whenever possible, you may try the new `distutils_enable_tests` API.
2. Test-deps go into BDEPEND.
3. Do all of those packages have working tests? Any of them need test support added?
4. Don't worry about Prefix keywords. Prefix team takes care of updating them as necessary. They may look weird, they combine their own overlay with ::gentoo.

arthurzam commented:
> Without reviewing yet, few notes:
> 
>     1. Whenever possible, you may try the new `distutils_enable_tests` API.

When changed to use this API, have added a message to commit.

>     2. Test-deps go into BDEPEND.
Done
>     3. Do all of those packages have working tests? Any of them need test support added?
Yes. For some of `dev-python/zope-*` packages I have added tests, and is documented in the commit message.
>     4. Don't worry about Prefix keywords. Prefix team takes care of updating them as necessary. They may look weird, they combine their own overlay with ::gentoo.

arthurzam commented:
Remove patches for `dev-python/{attrs,pluggy,hypothesis}`, as @mgorny had already bumped the versions.

mgorny commented on dev-python/cffi/cffi-1.13.2.ebuild:
```diff
@@ -0,0 +1,51 @@
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	virtual/libffi:=
```
This is a DEPEND, not BDEPEND.

mgorny commented on dev-python/cffi/cffi-1.13.2.ebuild:
```diff
@@ -0,0 +1,51 @@
+	dev-python/pycparser[${PYTHON_USEDEP}]"
+BDEPEND="${RDEPEND}
+	virtual/pkgconfig
+	doc? ( dev-python/sphinx )
```
I've screwed up here as it uses autodoc. Could you rewrite it to use the usually any_dep/check_deps voodoo? Or just rebase on mgorny:distutils-sphinx branch and use the new fancy API.

mgorny commented on dev-python/cffi/cffi-1.13.2.ebuild:
```diff
@@ -0,0 +1,51 @@
+	doc? ( dev-python/sphinx )
+	test? ( dev-python/pytest[${PYTHON_USEDEP}] )"
+
+# Avoid race on _configtest.c (distutils/command/config.py:_gen_temp_sourcefile)
```
I suppose this might no longer be necessary.

mgorny commented on dev-python/cffi/cffi-1.13.2.ebuild:
```diff
@@ -0,0 +1,51 @@
+}
+
+python_test() {
+	"${PYTHON}" -c "import _cffi_backend as backend" || die
```
EPYTHON here.

mgorny commented on dev-python/atomicwrites/atomicwrites-1.3.0.ebuild:
```diff
@@ -0,0 +1,47 @@
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~x86 ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="doc"
+
+# TODO: remove impl limiters when sphinx is py38
```
This is stale, please remove.

mgorny commented on dev-python/lazy-object-proxy/lazy-object-proxy-1.4.3.ebuild:
```diff
@@ -0,0 +1,38 @@
+}
+
+python_test() {
+	py.test -v -v --ignore=src || die "Fails for ${EPYTHON}"
```
Is this ignore really necessary? Maybe we could reuse `distutils_enable_tests` if not.

arthurzam replied:
This ignore is indeed necessary, without it it tries to `import cext` which fails on `python3`. On theirs testing CI, they also use `--ignore=src`.

mgorny commented on dev-python/pycryptodome/pycryptodome-3.9.4.ebuild:
```diff
@@ -0,0 +1,25 @@
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~ppc-aix ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~m68k-mint ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+
+RDEPEND="dev-libs/gmp:0
+	virtual/python-cffi[${PYTHON_USEDEP}]
```
This is BDEPEND, not DEPEND.

mgorny commented on dev-python/zope-interface/zope-interface-4.7.1.ebuild:
```diff
@@ -0,0 +1,46 @@
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-python/namespace-zope[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}"
```
Seems unnecessary.

mgorny commented on dev-python/zope-interface/zope-interface-4.7.1.ebuild:
```diff
@@ -0,0 +1,46 @@
+	distutils-r1_python_compile
+}
+
+distutils_enable_tests setup.py
```
Move it above phases.

mgorny commented on dev-python/zope-exceptions/zope-exceptions-4.3.ebuild:
```diff
@@ -0,0 +1,37 @@
+
+RDEPEND="dev-python/namespace-zope[${PYTHON_USEDEP}]
+	dev-python/zope-interface[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}"
```
Shouldn't be necessary.

mgorny commented on dev-python/zope-i18nmessageid/zope-i18nmessageid-5.0.0.ebuild:
```diff
@@ -0,0 +1,33 @@
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+
+RDEPEND="dev-python/six[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}"
```
Seems unnecessary.

mgorny commented on dev-python/zope-schema/zope-schema-4.9.3.ebuild:
```diff
@@ -0,0 +1,40 @@
+RDEPEND="dev-python/zope-event[${PYTHON_USEDEP}]
+	dev-python/zope-interface[${PYTHON_USEDEP}]"
+BDEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
+	test? ( ${RDEPEND}
```
d_e_t already takes care of that.

mgorny commented on dev-python/zope-deprecation/zope-deprecation-4.4.0.ebuild:
```diff
@@ -0,0 +1,32 @@
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="dev-python/namespace-zope[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}"
```
Unnecessary.

arthurzam commented:
Thanks for review - I hope I will know all in future so it will be easier.

1. Wasn't sure about `distutils_enable_tests setup.py` setting `IUSE` and `RESTRICT`. But after rechecking if `${RDEPEND}` isn't empty it sets it - nicely done :+1: 
2. Added `python3_8` to `pydiff` (needed for `autopep8`) - of course have checked with tests.

mgorny commented:
I'm going to push distutils_enable_sphinx shortly.

arthurzam commented:
Rebased over new `distutils_enable_sphinx`.
Needed to change only for `dev-python/{atomicwrites,cffi}`.
For every package I touch in this PR, fixed `QA: MissingTestRestrict` (watched through [here](https://qa-reports.gentoo.org/output/gentoo-ci/output.verbose.html)).

mgorny commented:
zope and imagesize done too. zope.schema doesn't run any tests.

arthurzam commented:
@mgorny , Thank you - Closing now.

1. Is it a good idea to add docs support for all python packages that have `sphinx` docs?
2. How should I add `python3_8` support for `dev-python/namespace-*` packages?

mgorny commented:
I didn't merge pyflakes & pycryptodome. Those two need fixes/rebase.

Namespace jus can be added when needed.
